Actions/VerifyPRChanges/VerifyPRChanges.ps1

```diff
@@ -6,25 +6,78 @@ Param(
   [Parameter(HelpMessage = "Head commit of the PR", Mandatory = $false)]
   [string] $headSHA,
   [Parameter(HelpMessage = "The name of the repository the PR is going to", Mandatory = $false)]
-  [string] $prBaseRepository
+  [string] $prBaseRepository,
+  [Parameter(HelpMessage = "The id of the pull request", Mandatory = $false)]
+  [string] $pullRequestId
 )
 
+function ValidateFiles
+(
+  [Object[]] $Files
+)
+{
+  $Files | ForEach-Object {
+    $filename = $_.filename
+    $status = $_.status
+    Write-Host "- $filename $status"
+    $extension = [System.IO.Path]::GetExtension($filename)
+    $name = [System.IO.Path]::GetFileName($filename)
+    if ($extension -eq '.ps1' -or $extension -eq '.yaml' -or $extension -eq '.yml' -or $name -eq "CODEOWNERS" -or $filename.StartsWith(".github/")) {
+      throw "Pull Request containing changes to scripts, workflows or CODEOWNERS are not allowed from forks."
+    }
+  } 
+}
+
+function ValidatePullRequest
+(
+  [string[]] $PullRequestRepository,
+  [string[]] $PullRequestId,
+  [object] $Headers,
+  [int] $MaxAllowedChangedFiles = 3000
+)
+{
+  $url = "https://api.github.com/repos/$($prBaseRepository)/pulls/$pullRequestId"
+  $pullRequestDetails = Invoke-WebRequest -UseBasicParsing -Headers $Headers -Uri $url | ConvertFrom-Json
+
+  # List Pull Request files has a max of 3000 files. https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests-files
+  if ($pullRequestDetails.changed_files -gt $MaxAllowedChangedFiles) {
+    throw "Pull request contains changes to $($pullRequestDetails.changed_files) files. You cannot change more than $MaxAllowedChangedFiles files from a fork."
+  }
+}
+
+function ValidatePullRequestFiles
+(
+  [string[]] $PullRequestRepository,
+  [string[]] $PullRequestId,
+  [object] $Headers
+)
+{
+  $pageNumber = 1
+  $resultsPerPage = 100
+  $hasMoreData = $true
+  Write-Host "Files Changed:"
+  while ($hasMoreData) {
+    $url = "https://api.github.com/repos/$($prBaseRepository)/pulls/$pullRequestId/files?per_page=$resultsPerPage&page=$pageNumber"
+    $changedFiles = Invoke-WebRequest -UseBasicParsing -Headers $headers -Uri $url | ConvertFrom-Json
+
+    ValidateFiles -Files $changedFiles
```

Comment:
What is $changedFiles if there are no changes in the PR?
Is it $null or is it an empty array?
if it is $null - then validateFiles will fail

maybe add @( ) around the invoke-webrequest  | ConvertFrom-Json to ensure an array or add 

`$Files | Where-Object { $_ } | ForEach-Object {`

in the function.

Reply:
Or maybe make ValidatePullRequest to a function returning true if it is necessary to validate the files (i.e. count -gt 0)

Reply:
I've added a check for this